### ISSUE TITLE
Submitting a markdown file produces strange output

### ISSUE THREAD

**Issue by keeferrourke** [ian-james/IFS#3]
I didn't have a regular text file on hand to submit for testing, so I uploaded a [README.md](https://github.com/keeferrourke/lang_tools/blob/master/README.md) and it dumped some javascript at the top of the text box. I have no idea what causes this.

FYI: `#Language Tools` is the beginning of the file.

```
$ event )" uib-popover-template="'components/Feedback/popoverMini.html'" popover-animation="true" popover-placement="bottom" class=" btnText spelling "> $ event )" uib-popover-template="'components/Feedback/popoverMini.html'" popover-animation="true" popover-placement="bottom" class=" btnText spelling "> $ event )" uib-popover-template="'components/Feedback/popoverMini.html'" popover-animation="true" popover-placement="bottom" class=" btnText spelling "> # Language Tools

This set of scripts will parse text and provide suggestions for spelling
and grammatical corrections. The scripts may be used on the command line with
verbose console output, or to produce JSON  formatted output.
```

thoughts?

**Comment by ian-james** [comment]
Haven't directly looked into this but if you're seeing that code at the top it means that the highlighter tried to replace words (likely to many words) near the same location.

The popoverMini is the onClick popover that happens when an incorrect word is selected.

**Comment by ian-james** [comment]
This might be fixed via file submission properties but not if they have a text file that is actually markdown.
To confirm.

**Comment by ian-james** [comment]
Apparently, we can still submit .md files and it works for some tools.

Language Check throws this error

TD error: Traceback (most recent call last):
  File "./tools/languageTools/grammar/grammar.py", line 241, in <module>
    main(sys.argv[1:])
  File "./tools/languageTools/grammar/grammar.py", line 224, in main
    json_data = build_json(infile, lang, matches)
  File "./tools/languageTools/grammar/grammar.py", line 101, in build_json
    json_obj = json.loads(json_string)
  File "/usr/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.5/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Invalid \escape: line 11 column 271 (char 1932)

**Comment by keeferrourke** [comment]
I'll take a look at the language check tool tonight. That shouldn't happen, and I don't think that error ever occurred on the Python 2 versions of the utilities. Will report back after investigation!

**Comment by keeferrourke** [comment]
grammar.py was choking when it found an unescaped backslash character in the context string. this has been fixed as of commit: 389a59a170fbd87acf3e366acaaba90cf50cf7bd

I think this issue can be closed; but if we have more issues with parsing strange characters, it should be opened again.
